internal/resources/syntheticmonitoring/resource_probe_test.go

```diff
@@ -94,6 +94,64 @@ func TestAccResourceProbe_recreate(t *testing.T) {
 	})
 }
 
+// Test that a probe that is used in a check can be deleted or recreated
+func TestAccResourceProbe_recreateProbeUsedInCheck(t *testing.T) {
+	testutils.CheckCloudInstanceTestsEnabled(t)
+
+	randomName := acctest.RandomWithPrefix("tf")
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testSyntheticMonitoringProbeAndCheck(randomName, "test1"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("grafana_synthetic_monitoring_probe.first", "id"),
+					resource.TestCheckResourceAttrSet("grafana_synthetic_monitoring_probe.second", "id"),
+					resource.TestCheckResourceAttrSet("grafana_synthetic_monitoring_check.http", "id"),
+					resource.TestCheckResourceAttr("grafana_synthetic_monitoring_probe.first", "name", randomName+"test1-first"),
+					resource.TestCheckResourceAttr("grafana_synthetic_monitoring_probe.second", "name", randomName+"test1-second"),
+				),
+			},
+			// Change the name of the probe
```

Comment:
I'm not familiar with Terraform providers implementation, so just to make sure.. I understand this test verifies that the check that was originally associated with both test probes (_first_ and _second_), is now associated with the "new" probes after the rename. Is that correct?

Reply:
It checks that we can rename used probes, whether or not it recreates them. The original issue mentioned that renaming probes was an issue